internal/dag/httpproxy_processor.go

```diff
@@ -516,6 +516,48 @@ func (p *HTTPProxyProcessor) computeHTTPProxy(proxy *contour_api_v1.HTTPProxy) {
 	}
 	insecure.CORSPolicy = cp
 
+	var (
+		requestHeadersPolicy  HeadersPolicy
+		responseHeadersPolicy HeadersPolicy
+	)
+	requestHeadersPolicyDefined := false
+	responseHeadersPolicyDefined := false
+	if proxy.Spec.VirtualHost.RequestHeadersPolicy != nil {
```

Comment:
I think we will want to reuse the existing logic we have for processing these HeaderPolicies - in particular e.g. [headersPolicyRoute](https://github.com/projectcontour/contour/blob/7feb49e86b3598e78421f6afe249ff6740cd6d9b/internal/dag/policy.go#L168). That function has some better validation and also handles parsing dynamic header values (see https://projectcontour.io/docs/1.26/config/request-rewriting/#dynamic-header-values), which I think we should support for vhost header policies as well. We'd want to pass `false` as the second argument since we won't support rewriting the Host header for vhosts.

Reply:
Done